pkg/convert/arb.go

```diff
@@ -0,0 +1,107 @@
+package convert
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+	"go.expect.digital/translate/pkg/model"
+)
+
+/* .arb files employ the key-value (key-translation) format, with separate files representing different languages.
+
+app_en.arb
+{
+	"title": "Hello World!",
+	"@title" : {
+		"description" : "Message to greet the World"
+	},
+	"greeting": "Welcome {user}!",
+	"@greeting": {
+		"placeholders": {
+			"user":{
+				"type":"string",
+				"example":"Bob"
+			}
+		}
+	}
+}
+
+app_fr.arb
+{
+  "title": "Bonjour le monde!",
+  "greeting": "Bienvenue {user}!"
+}
+
+*/
+
+// Converts a serialized data in ARB file format into model.Messages.
+func FromArb(data []byte) (model.Messages, error) {
+	var dst map[string]interface{}
+	if err := json.Unmarshal(data, &dst); err != nil {
+		return model.Messages{}, fmt.Errorf("unmarshal from ARB to model.Messages: %w", err)
+	}
+
+	findDescription := func(key string) (string, error) {
+		// if '@' prefix missing then no additional information is provided
+		subKeyMap, ok := dst["@"+key]
+		if !ok {
+			return "", nil
+		}
+
+		var meta struct {
+			Description string `json:"description"`
+		}
+
+		if err := mapstructure.Decode(subKeyMap, &meta); err != nil {
+			return "", fmt.Errorf("decode subKeyMap to meta: %w", err)
+		}
+
+		return meta.Description, nil
+	}
+
+	var messages model.Messages
+
+	for key, value := range dst {
+		// Ignore a key if it begins with '@' as it only supplies additional information and not the actual message.
+		if key[0] == '@' {
+			continue
+		}
+
+		msg := model.Message{ID: key}
+
+		// If a key does not have an '@' prefix and its value is not of type string, then file is not formatted correctly.
+		var ok bool
+		if msg.Message, ok = value.(string); !ok {
+			return model.Messages{}, fmt.Errorf("unsupported value type '%T' for key '%s'", value, key)
+		}
+
+		var err error
+		if msg.Description, err = findDescription(key); err != nil {
+			return model.Messages{}, fmt.Errorf("find description of '%s': %w", key, err)
+		}
+
+		messages.Messages = append(messages.Messages, msg)
+	}
+
+	return messages, nil
+}
+
+// Converts model.Messages into a serialized data in ARB file format.
+func ToArb(messages model.Messages) ([]byte, error) {
+	dst := make(map[string]interface{}, len(messages.Messages)*2) //nolint:gomnd
```

Comment:
Due to additional metadata keys, the map size is at least `len(messages.Messages)` and the max size of a map can be up to `len(messages.Messages)*2`.
It is better to allocate a max size beforehand, afaik